nautobot/core/settings_funcs.py

```diff
@@ -82,3 +84,84 @@ def parse_redis_connection(redis_database):
         return f"{redis_scheme}://{redis_creds}{redis_host}?db={redis_database}"
     else:
         return f"{redis_scheme}://{redis_creds}{redis_host}:{redis_port}/{redis_database}"
+
+
+def setup_structlog_logging(
+    django_logging: dict,
+    django_apps: list,
+    django_middleware: list,
+    log_level="INFO",
+    root_level="INFO",
+    debug=False,
+    debug_db=False,
+    plain_format=False,
+) -> None:
+    """Set up structlog logging for Django."""
+    django_logging["version"] = 1
+    django_logging["disable_existing_loggers"] = True
+    if "test" in sys.argv:
+        django_logging["handlers"] = {
+            "null_handler": {
+                "level": "INFO",
+                "class": "logging.NullHandler",
+            },
+        }
+        for logger in django_logging["loggers"].values():
+            logger["handlers"] = ["null_handler"]
+            logger["level"] = "INFO"
+
+    django_apps.append("django_structlog")
+    django_middleware.append("django_structlog.middlewares.RequestMiddleware")
+
+    processors = (
+        structlog.stdlib.add_log_level,
+        structlog.stdlib.add_logger_name,
+        structlog.stdlib.PositionalArgumentsFormatter(),
+        structlog.contextvars.merge_contextvars,
+        structlog.processors.TimeStamper(fmt="iso"),
+        structlog.processors.StackInfoRenderer(),
+        structlog.processors.UnicodeDecoder(),
+        *([] if debug else [structlog.processors.format_exc_info]),
```

Comment:
Would be good to have comments explaining the basic purpose/effect of each of these.

Reply:
Added comments.